### ISSUE TITLE
Dotnet format should take a single file

### ISSUE THREAD

**Issue by IanGraingerGMSL** [dotnet/sdk#43056]
### Description of the problem

Dotnet format is fantastic. We use it to check all the .editorconfig rules have been followed in our builds.

But really I want a version of prettier that I have for Typescript. So that I can run dotnet format on the open file every time I press 'Save', and so I could run it on just modified files as a pre-commit hook (rather than having to run it on the whole solution).

### Describe the solution I'd like
`dotnet format [filename|filename list|file glob pattern]`.

**Comment by Mattwmaster58** [comment]
Somebody mentioned the `--include` flag - I don't think this is good enough, it still requires you to specify a project/solution. I'm naive to the reason for this, but I assume it's to help with import solution or something like that. If that's the case, that doesn't seem like a _great_ reason. prettier for example can operator on a single tsx/jsx without knowledge of the rest of the project as far as I'm aware, and I think it would be a great feature to have with dotnet format, too

**Comment by x0rld** [comment]
> Somebody mentioned the `--include` flag - I don't think this is good enough, it still requires you to specify a project/solution. I'm naive to the reason for this, but I assume it's to help with import solution or something like that. If that's the case, that doesn't seem like a _great_ reason. prettier for example can operator on a single tsx/jsx without knowledge of the rest of the project as far as I'm aware, and I think it would be a great feature to have with dotnet format, too

I tried the --include, it scan all the solution then do your file 
for example on a project I have it take 2 minutes with one file include cause it scan all the files

**Comment by ChadNedzlek** [comment]
I was shocked how slow "dotnet format" was, even for tiny little files. It feels like it should be near instantaneous. But yeah, it's so horribly slow that I basically just get constant CI breaks because I can't reliably run it quickly on edits, and I always forget to run it before pushing.

**Comment by Frulfump** [comment]
> I was shocked how slow "dotnet format" was, even for tiny little files. It feels like it should be near instantaneous. But yeah, it's so horribly slow that I basically just get constant CI breaks because I can't reliably run it quickly on edits, and I always forget to run it before pushing.

Yeah! See this issue as well https://github.com/dotnet/sdk/issues/49910
